lib/charms/karma_k8s/v0/karma.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+""" # Karma library
+
+This library is designed to be used by a charm consuming or providing the karma-dashboard relation.
+"""
+
+import logging
+
+import ops.charm
+from ops.framework import EventBase, EventSource, ObjectEvents
+from ops.charm import RelationJoinedEvent, RelationDepartedEvent
+from ops.relation import ConsumerBase, ProviderBase
+from ops.framework import StoredState
+
+from typing import List, Dict, Optional, Iterable
+
+# The unique Charmhub library identifier, never change it
+LIBID = "abcdef1234"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 2
+
+logger = logging.getLogger(__name__)
+
+
+class KarmaAlertmanagerConfig:
+    required_fields = ("name", "uri")
+
+    @staticmethod
+    def is_valid(config: Dict[str, str]):
+        return all(key in config for key in KarmaAlertmanagerConfig.required_fields)
+
+
+def dict_subset(data: dict, key_subset: Iterable):
```

Comment:
Only used once. Feel free to keep, but could be a lambda or just a comprehension instead of making people read it look up a single use function

Reply:
Good point. Removed.
BTW: `PEP 8: E731 do not assign a lambda expression, use a def`